cmd/ndm_daemonset/probe/addhandler.go

```diff
@@ -336,49 +338,68 @@ func (pe *ProbeEvent) deviceInUseByMayastor(bd blockdevice.BlockDevice, bdAPILis
 // deviceInUseByZFSLocalPV check if the device is in use by zfs localPV and returns true if further processing of
 // event is required. If the device has ZFS pv on it, then a blockdevice resource will be created and zfs PV tag
 // will be added on to the resource
-func (pe *ProbeEvent) deviceInUseByZFSLocalPV(bd blockdevice.BlockDevice, bdAPIList *apis.BlockDeviceList) (bool, error) {
+func (pe *ProbeEvent) deviceInUseByZFSLocalPV(bd blockdevice.BlockDevice, bdAPIList *apis.BlockDeviceList) (*apis.BlockDevice, bool, error) {
 	if bd.DeviceAttributes.DeviceType == blockdevice.BlockDeviceTypePartition {
 		parentBD, ok := pe.Controller.BDHierarchy[bd.DependentDevices.Parent]
 		if !ok {
 			klog.Errorf("unable to find parent device for %s", bd.DevPath)
-			return false, fmt.Errorf("error in getting parent device for %s from device hierarchy", bd.DevPath)
+			return nil, false, fmt.Errorf("error in getting parent device for %s from device hierarchy", bd.DevPath)
 		}
 		if parentBD.DevUse.InUse && parentBD.DevUse.UsedBy == blockdevice.ZFSLocalPV {
 			klog.V(4).Infof("ParentDevice: %s of device: %s in use by zfs-localPV", parentBD.DevPath, bd.DevPath)
-			return false, nil
+			return nil, false, nil
 		}
 
 	}
 	if !bd.DevUse.InUse {
-		return true, nil
+		return nil, true, nil
 	}
 
 	// not in use by zfs localpv
 	if bd.DevUse.UsedBy != blockdevice.ZFSLocalPV {
-		return true, nil
+		return nil, true, nil
 	}
 
 	klog.Infof("device: %s in use by zfs-localPV", bd.DevPath)
 
 	uuid, ok := generateUUIDFromPartitionTable(bd)
 	if !ok {
 		klog.Errorf("unable to generate uuid for zfs-localPV device: %s", bd.DevPath)
-		return false, fmt.Errorf("error generating uuid for zfs-localPV disk: %s", bd.DevPath)
+		return nil, false, fmt.Errorf("error generating uuid for zfs-localPV disk: %s", bd.DevPath)
 	}
 
 	bd.UUID = uuid
 
 	deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
 	bdAPI := deviceInfo.ToDevice()
+	patch := client.MergeFrom(bdAPI.DeepCopy())
 	bdAPI.Labels[kubernetes.BlockDeviceTagLabel] = string(blockdevice.ZFSLocalPV)
+	bdAPI.Namespace = pe.Controller.Namespace
+
+	err := pe.Controller.Clientset.Create(context.TODO(), bdAPI.DeepCopy(), &client.CreateOptions{
+		FieldManager: "NDM",
+	})
+	if err == nil {
+		klog.Infof("eventcode=%s msg=%s rname=%v",
+			"ndm.blockdevice.create.success", "Created blockdevice object in etcd",
+			bdAPI.ObjectMeta.Name)
+		return &bdAPI, false, err
+	}
 
-	err := pe.Controller.CreateBlockDevice(bdAPI)
```

Comment:
Why is this change required? The blockdevices as part of ZFS localpv will only be updated by NDM. So there is no chance of conflicts there.